packages/frontend/src/redux/slices/recoveryMethods/index.js

```diff
@@ -0,0 +1,105 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import set from 'lodash.set';
+import { createSelector } from 'reselect';
+
+import { wallet } from '../../../utils/wallet';
+import createParameterSelector from '../createParameterSelector';
+import initialErrorState from '../initialErrorState';
+
+const SLICE_NAME = 'recoveryMethods';
+
+const initialState = {
+    byAccountId: {}
+};
+
+const initialAccountIdState = {
+    items: [],
+    status: {
+        loading: false,
+        error: initialErrorState
+    }
+};
+
+const fetchRecoveryMethods = createAsyncThunk(
+    `${SLICE_NAME}/fetchRecoveryMethods`,
+    async ({ accountId }, thunkAPI) => {
+        const { dispatch } = thunkAPI;
+
+        const recoveryMethods = await wallet.getRecoveryMethods();
+
+        const { actions: { setRecoveryMethods } } = recoveryMethodsSlice;
+        dispatch(setRecoveryMethods({ recoveryMethods, accountId }));
+    },
+    {
+        condition: ({ accountId }, thunkAPI) => {
+            const { getState } = thunkAPI;
+            if (selectRecoveryMethodsLoading(getState(), { accountId })) {
+                return false;
+            }
+        }
+    }
+);
+
+const recoveryMethodsSlice = createSlice({
+        name: SLICE_NAME,
+        initialState,
+        reducers: {
+            setRecoveryMethods(state, { payload }) {
+                const { recoveryMethods, accountId } = payload;
+                set(state, ['byAccountId', accountId, 'items'], recoveryMethods);
+            }
+        },
+        extraReducers: ((builder) => {
+            builder.addCase(fetchRecoveryMethods.pending, (state, { meta }) => {
+                const { accountId } = meta.arg;
+    
+                set(state, ['byAccountId', accountId, 'status', 'loading'], true);
+                set(state, ['byAccountId', accountId, 'status', 'error'], initialErrorState);
+            });
+            builder.addCase(fetchRecoveryMethods.fulfilled, (state,  { meta }) => {
+                const { accountId } = meta.arg;
+    
+                set(state, ['byAccountId', accountId, 'status', 'loading'], false);
+                set(state, ['byAccountId', accountId, 'status', 'error'], initialErrorState);
+            });
+            builder.addCase(fetchRecoveryMethods.rejected, (state, { meta, error }) => {
+                const { accountId } = meta.arg;
+                
+                set(state, ['byAccountId', accountId, 'status', 'loading'], false);
+                set(state, ['byAccountId', accountId, 'status', 'error'], {
+                    message: error?.message || 'An error was encountered.',
+                    code: error?.code
+                });
+            });
+        })
+    }
+);
```

Comment:
Not a blocker for this PR, but since this is a pretty standard set of handlers, maybe we can make this a helper... what do you think about something like:

```javascript
// This in a utility file somewhere
function addAsyncThunkHandlers({ asyncThunk, builder, buildStatusPath }) {
    builder.addCase(asyncThunk.pending, (state, action) => {
        set(state, [...buildStatusPath(action), 'loading'], true);
        set(state, [...buildStatusPath(action), 'error'], initialErrorState);
    });
    builder.addCase(asyncThunk.fulfilled, (state, action) => {
        set(state, [...buildStatusPath(action), 'loading'], false);
        set(state, [...buildStatusPath(action), 'error'], initialErrorState);
    });
    builder.addCase(asyncThunk.rejected, (state, action) => {
        const { error } = action;
        set(state, [...buildStatusPath(action), 'loading'], false);
        set(state, [...buildStatusPath(action), 'error'], {
            message: error?.message || 'An error was encountered.',
            code: error?.code
        });
    });
}

const recoveryMethodsSlice = createSlice({
        name: SLICE_NAME,
        initialState,
        reducers: {
            setRecoveryMethods(state, { payload }) {
                const { recoveryMethods, accountId } = payload;
                set(state, ['byAccountId', accountId, 'items'], recoveryMethods);
            }
        },
        extraReducers: ((builder) => {
          addLoadingLifecycle({
            asyncThunk: fetchRecoveryMethods,
            builder,
            buildStatusPath: ({ meta: { arg: { accountId } } }) => (['byAccountId', accountId, 'status'])
          });
        })
    }
);
```

Do you think that would be configurable enough for our current pattern?

Reply:
Yes, I was thinking about a similar solution, it will be super handy if we want to use the same `status` object for all slices. I added an issue for this change #2182 , I will implement it after finishing with selectors, so the selectors would be ready to go quickly.